app/models/piece.rb

```diff
@@ -32,4 +32,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(new_x, new_y)
+    @target = Piece.where(:x_position => new_x).where(:y_position => new_y).take
```

Comment:
Will need to specify the game for the piece that is calling this method. One option is to change this code to:  @target = self.game.pieces.where(:x_position => new_x, :y_position => new_y).take          (you can combine the two where method calls into one to be slightly more efficient with a single database query)